apps/dash-daq-tektronix350/app.py

```diff
@@ -0,0 +1,542 @@
+                        [
+                            daq.PowerButton(
+                                id="oscilloscope",
+                                on=cur_inputs[cur_tab]["oscilloscope"],
+                                label="Oscilloscope",
+                                labelPosition="bottom",
+                                color=theme["primary"],
+                            )
+                        ],
+                        className="six columns",
+                    ),
+                ],
+                style={"margin": "15px 0"},
+            ),
+        ],
+    )
+
+
+def function_setting_div(cur_input, cur_tab):
+    if cur_input is None or len(cur_input) == 0:
+        cur_input = init_input
+    return html.Div(
+        className="row power-settings-tab",
+        children=[
+            html.Div(
+                className="Title",
+                style={"color": theme["primary"]},
+                children=html.H3("Function", id="function-title"),
+            ),
+            html.Div(
+                children=[
+                    # Knobs
+                    knobs(cur_input, cur_tab),
+                    # LED Displays
+                    led_displays(cur_input, cur_tab),
+                    # # RadioItems
+                    radioitem(cur_input, cur_tab),
+                ]
+            ),
+        ],
+    )
+
+
+# Main App
+app.layout = html.Div(
+    id="main-page",
+    children=[
+        # Header
+        html.Div(
+            id="header",
+            className="banner row",
+            children=[
+                # Logo and Title
+                html.Div(
+                    className="banner-logo-and-title",
+                    children=[
+                        html.Img(
+                            src=app.get_asset_url("dash-logo-white.png"),
+                            className="logo",
+                        ),
+                        html.H2(
+                            "Dash DAQ: Function Generator & Oscilloscope Control Panel"
+                        ),
+                    ],
+                ),
+                # Toggle
+                html.Div(
+                    className="row toggleDiv",
+                    children=daq.ToggleSwitch(
+                        id="toggleTheme", size=40, value=False, color="#00418e"
+                    ),
+                ),
+            ],
+        ),
+        html.Div(
+            className="row",
+            children=[
+                # LEFT PANEL - SETTINGS
+                html.Div(
+                    className="five columns",
+                    children=[
+                        html.Div(
+                            id="left-panel",
+                            children=[
+                                html.Div(
+                                    id="dark-theme-components",
+                                    className="left-panel-controls",
+                                    children=DarkThemeProvider(
```

Comment:
I'm not too familiar with how DarlThemeProvider works, why is there only one call and why is it only applied in the left panel?

Reply:
Dark theme provider only applies dark theme to the `dash-daq` components in the app. The left panel is the only part with `daq` components so that's why it's only on the left. The rest is colored using callbacks.